system/datatypes/peer_reviewed_article.php

```diff
@@ -576,262 +239,235 @@ public function indexAction() {
-					if (self::FIELD_NAME_DISCUSSION_REVIEW_CONTAINER !== $type && isUser()) {
-						$committerEmail = $comment->getAttr(self::FIELD_NAME_DISCUSSION_COMMENTER_EMAIL);
-						$html .= ' <span> | ' . __('art-email') . ': <a href="mailto:' . htmlspecialchars($committerEmail) . '">' . htmlspecialchars($committerEmail) . '</a></span>';
-					}
-					if ($firstComment) {
-						if ($blocking) {
-							$html .= ' | ' . ($resolved ? __('art-is-resolved') : __('art-is-blocking'));
-						}
-					}
-					$html .= '</p>';
-					$list->append('<li ' . ($firstComment ? 'class="first-comment"' : '') . '>' . $html . '</li>');
-					$firstComment = false;
-				}
-			}
-
-			if (!$list->hasChildNodes()) {
-				// if there are no comments to be displayed, all discussion parts do not need to be displayed.
-				continue;
-			}
+		$this->appendEditForm($form);
+		return null;
+	}
 
-			/** @var HyphaDomElement $reviewCommentContainer */
-			$reviewCommentContainer = $this->xml->createElement('div');
-			$class = 'review-comment-wrapper collapsed';
-			foreach (['blocking' => $blocking, 'resolved' => $resolved, 'closed' => $closed] as $name => $isTrue) {
-				if ($isTrue) {
-					$class .= ' ' . $name;
-				}
-			}
-			$reviewCommentContainer->attr('class', $class);
-			$reviewCommentContainer->append($list);
+	/**
+	 * Updates the article with the posted data.
+	 *
+	 * @param HyphaRequest $request
+	 * @return array|null
+	 */
+	public function editAction(HyphaRequest $request) {
+		if (!isUser()) {
+			notify('error', __('login-to-perform-action'));
+			return ['redirect', $this->constructFullPath($this->pagename)];
+		}
 
-			if (self::FIELD_NAME_DISCUSSION_REVIEW_CONTAINER === $type) {
-				$msgId = null;
-				if ($blocking) {
-					if (!$resolved && ($discussion->getAttr(self::FIELD_NAME_USER) === $this->hyphaUser->getAttribute('id') || isAdmin())) {
-						$msgId = __('art-set-as-resolved');
-					}
-				} elseif (!$closed) {
-					$msgId = __('art-set-as-closed');
-				}
-				if (null !== $msgId) {
-					$path = str_replace('{id}', $discussion->getId(), self::PATH_DISCUSSION_CLOSED);
-					$list->append('<p>' . $this->makeActionButton(__($msgId), $path, self::FORM_CMD_DISCUSSION_CLOSED) . '</p>');
-				}
-			}
+		// create form
+		$form = $this->createEditForm($request->getPostData());
 
-			// display comment form if the discussion is still open
-			if (!$closed) {
-				$replyForm = $this->createDiscussionCommentForm($discussion);
-				$list->append($replyForm);
-			}
+		// process form if there are no errors
+		if (empty($form->errors)) {
+			$this->xml->lockAndReload();
+			$author = $form->dataFor(self::FIELD_NAME_AUTHOR);
+			$this->xml->find(self::FIELD_NAME_CONTEXT)->setAttr(self::FIELD_NAME_AUTHOR, $author);
+			$this->xml->find(self::FIELD_NAME_TITLE)->setText($form->dataFor(self::FIELD_NAME_TITLE));
+			$this->xml->find(self::FIELD_NAME_TEXT)->setHtml($form->dataFor(self::FIELD_NAME_TEXT), true);
```

Comment:
This setHTML call seems a bit opaque. I think the `true` is intended to wikify, but that should probably be commented, or perhaps use a new `setWikifiedHtml()` method? Also, it seems that `setHTML()` with two arguments was added by `DocPage`, which is deleted by this commit? Does this code actually work right now?

Reply:
fixed